plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -522,12 +523,12 @@ def process_main(self, item=None):
                                                       REPO_DATA_DIR_NAME)
 
         metadata_file_name = os.path.basename(unit._storage_path)
-        link_path = os.path.join(publish_location_relative_path, metadata_file_name)
-        plugin_misc.create_symlink(unit._storage_path, link_path)
+        file_path = os.path.join(publish_location_relative_path, metadata_file_name)
+        shutil.copyfile(unit._storage_path, file_path)
```

Comment:
Any reason to use specifically `shutil.copyfile` and not `shutil.copy`?
My understanding that `shutil.copy` copies also permission bits (to the extent possible), which is probably useful in our case.
Unlike `shutil.copy2`, `shutil.copy` doesn't copy all the file metadata, so creation/access time should be correct and not copied from the old file.

Reply:
I chose shutil.copyfile because I didn't think there was any value in preserving any attributes. I gave this a lot of thought and couldn't think of a reason to preserve them. Both files are created by a celery worker running as user 'apache'.

Reply:
Isn't the fact that the file could be rsynced by rpm_rsync_distributor a reason to preserve attributes?

rsync will do "quick check" by default - if file has same path, size and mtime, then rsync will skip checksumming the file on both sides.

If mtime changes on every publish even though file content didn't change, that means rsync will be forced to read the file on the remote each time to determine whether it changed.

Reply:
You bring up a very good point. I'll make a second PR.

Reply:
https://github.com/pulp/pulp_rpm/pull/1307 ... already merged.